### PR TITLE
Draft: Remove `eth` protocol changes.

### PR DESCRIPTION
**Pull request by palango**
### Description

For gathering feedback for now.

This removes a change to the ethereum protocol in our code, in which we send the block hashes with the bodies, for which I see no need.

If we decide to merge this, we need to create a new protocol version for it.

### CODE REVIEW

File: eth/fetcher/block_fetcher.go [hunk 1]
```diff
@@ -596,16 +595,14 @@ func (f *BlockFetcher) loop() {
...
+				for i := 0; i < len(task.transactions) && i < len(task.randomness) && i < len(task.epochSnarkData); i++ {
 					// Match up a body to any possible completion request
 					var matched = false
 					for hash, announce := range f.completing {
 						if f.queued[hash] != nil || announce.origin != task.peer {
 							continue
 						}
-						if task.blockHashes[i] != announce.header.Hash() {
-							continue
-						}
```
Comment (palango) [inline eth/fetcher/block_fetcher.go:608]: @mcortesi @gastonponti This seems to be the fundamental change that passing the hashes allows for. Do you see any reason for that?

Comment (carterqw2) [reply eth/fetcher/block_fetcher.go:608]: Looks like this was a [part](https://github.com/celo-org/celo-blockchain/pull/952) of a [bug](https://github.com/celo-org/celo-blockchain/issues/950) fix. I'm worrying that authors haven't added a test that tests this situation, reverting it looks risky, no?

Comment (palango) [reply eth/fetcher/block_fetcher.go:608]: Sure, but this is something we need to fix anyways when we want to separate protocols.

The only explanation that I have is that we could mistake an uncle block for  the correct one a take the wrong chain?
```go
if task.blockHashes[i] != announce.header.Hash() {
	continue
}
```

**Comment by palango** [comment]
This cannot be removed right now.

**Pull request closed without merging**